src/pages/AdminPages/ApprovedAccounts/index.js

```diff
@@ -0,0 +1,54 @@
+import React, {useEffect} from 'react'
+import { useSelector, useDispatch, shallowEqual } from 'react-redux'
+import Authenticate from '../../../components/Authenticate'
+import PageNotFound from '../../PageNotFound'
+import { 
+    getTraders, getUsers, getAdmins,
+    getAffiliates, setAlterUsers 
+} from "../../../store/reducers/adminQuery";
+import AlterUsers from '../../../components/Admin/AlterUsers';
+import { setIsLoading } from '../../../store/reducers/user';
+
+const ApprovedAccounts = () => {
+    const dispatch = useDispatch()
+    const { role, isLoading, alterUsers, traders } = useSelector(state => ({
+        role : state.user.role,
+        isLoading : state.user.isLoading,
+        alterUsers: state.adminQuery.alterUsers,
+        traders: state.adminQuery.traders,
+    }), shallowEqual)
+
+    useEffect(() => {
+        dispatch(setIsLoading());
+        dispatch(getUsers())
+        dispatch(getTraders())
+        // eslint-disable-next-line
+    }, [])
+
+    const update = (dispatchFunc) => {
+        dispatch(setIsLoading());
+        dispatch(dispatchFunc())
+    }
+
+    return(
+        <>
+            <Authenticate inside={true} />
+            {
+                role && (
+                    role.isAdmin ? (
+                    <>
+                        <button disabled={alterUsers[0] && alterUsers[0].role.isUser && !alterUsers[0].role.isAffiliate  && true} onClick={() => {update(getUsers)}}>View User Accounts</button> <br />
```

Comment:
Then what happens if the first entry in the array is an affiliate even though we are trying to get users who can also be affiliates?

Reply:
One of the main reason why I decided to disable the button is to know which button was clicked and which data is being rendered, but now I'd just create a new column for the roles so the admin knows which data is being rendered.